### ISSUE TITLE
What's the proper config to debug pytest?

### ISSUE THREAD

**Issue by ztlevi** [emacs-lsp/dap-mode#147]
Vscode config to test pytest like this is working.
```
    {
      "name": "PyTest test_app",
      "type": "python",
      "request": "launch",
      "module": "pytest",
      "args": ["-sv"],
      "program": "${workspaceFolder}/tests/test_app.py",
      "console": "integratedTerminal"
    },
```
Here is the output:
```
➜ cd /Users/ztlevi/Developer/Github/pytest_tutorial ; env PYTHONIOENCODING=UTF-8 PYTHONUNBUFFERED=1 /usr/local/bin/python3 /Users/ztlevi/.vscode/extensions/ms-python.python-2019.10.44104/pythonFiles/ptvsd_launcher.py --default --client --host localhost --port 62936 -m pytest -sv 
=================================================================================================================== test session starts ====================================================================================================================
platform darwin -- Python 3.7.5, pytest-5.2.2, py-1.8.0, pluggy-0.13.0 -- /usr/local/opt/python/bin/python3.7
cachedir: .pytest_cache
rootdir: /Users/ztlevi/Developer/Github/pytest_tutorial
collected 6 items                                                                                                                                                                                                                                          

tests/test_app.py::test_validate_good PASSED
tests/test_app.py::test_bad_file PASSED
tests/test_app.py::test_bad_json PASSED
tests/test_app.py::test_get_json PASSED
tests/test_app.py::test_get_validation [1]    79351 terminated  env PYTHONIOENCODING=UTF-8 PYTHONUNBUFFERED=1 /usr/local/bin/python3    --hos
```
But I tried something like this but no luck to get it working.
```
(dap-register-debug-template
 "My App test"
 (list :type "python"
       :request "launch"
       :args "-m pytest -sv"
       :target-module "tests/test_app.py"
       :name "My App test"))
```

It doesn't seem like dap-python pass `:program` and `:module` to the plist. What's the proper config to run pytest?

**Comment by ztlevi** [comment]
BTW, I don't have the UI debugger panes pop up. Was it implemented for python?

**Comment by yyoncho** [comment]
Please, provide a sample project I will try to make it work(+ additional setup steps if needed). Note that I am not familiar with python and this will help me a lot.

**Comment by yyoncho** [comment]
Fixed with 7eea7ba . Thank you for the detailed instructions on how to reproduce the issue, it really helped me with fixing it.

**Comment by saltsucker** [comment]
Does this feature allow for passing arguments to `pytest`?
Relevant question: https://github.com/emacs-lsp/dap-mode/pull/274#issuecomment-1211427276
